runtime/src/accounts_db.rs

```diff
@@ -59,6 +59,8 @@ pub struct ErrorCounters {
     pub missing_signature_for_fee: usize,
 }
 
+pub type CreditOnlyLocks = Vec<Arc<AtomicU64>>;
```

Comment:
Doesn't seem like there's a need for CreditOnlyLocks anymore, we can just return a single Results vector from lock_accounts(), and then on unlock, decrement the Credit-only lock count for the credit-only accounts in the transaction